instrumentation/opentelemetry-instrumentation-aiohttp-server/src/opentelemetry/instrumentation/aiohttp_server/__init__.py

```diff
@@ -0,0 +1,174 @@
+    ExcludeList,
+    get_excluded_urls,
+    get_traced_request_attrs,
+)
+
+from .version import __version__
+
+RequestHook = t.Callable[[trace.Span, BaseRequest], t.Any]
+ResponseHook = t.Callable[[trace.Span, BaseRequest, StreamResponse], t.Any]
+
+
+_CONFIG_PREFIX = "AIOHTTP_SERVER"
+_CONFIG_TRACED_REQUEST_ATTRS = get_traced_request_attrs(_CONFIG_PREFIX)
+_CONFIG_EXCLUDED_URLS = get_excluded_urls(_CONFIG_PREFIX)
+_SPAN_KEY = "opentelemetry-instrumentor-aiohttp-server.span_key"
+
+
+def _span_name(request: BaseRequest) -> str:
+    return f"HTTP {request.method.upper()}"
+
+
+def _is_suppressed(excluded_urls: ExcludeList, request: BaseRequest) -> bool:
+    return context.get_value(
+        "suppress_instrumentation"
+    ) or excluded_urls.url_disabled(str(request.url))
+
+
+def _collect_request_attributes(
+    request: BaseRequest, url_filter: t.Callable[[str], str]
+) -> t.Dict[str, str]:
+    url = str(request.url)
+    if url_filter:
+        url = url_filter(url)
+
+    return {
+        SpanAttributes.HTTP_METHOD: request.method.upper(),
+        SpanAttributes.HTTP_URL: url,
+        SpanAttributes.HTTP_FLAVOR: ".".join(map(str, request.version)),
+    }
+
+
+def _set_span_status(span: trace.Span, response: StreamResponse):
+    if not span.is_recording():
+        return
+
+    reason = None
+    try:
+        http_status = int(response.status)
+    except ValueError:
+        http_status = response.status
+        reason = f"Unable to parse status code: {http_status!r}"
+    except AttributeError:
+        reason = "Response without a status code"
+
+    if reason:
+        span.set_status(trace.Status(trace.StatusCode.ERROR), reason=reason)
+    else:
+        span.set_status(trace.Status(http_status_to_status_code(http_status)))
+        span.set_attribute(SpanAttributes.HTTP_STATUS_CODE, http_status)
+
+
+def _identity(arg):
+    return arg
+
+
+def _instrument(
+    *,
+    url_filter: t.Callable[[str], str] = _identity,
+    excluded_urls: t.Optional[t.List[str]] = None,
+    traced_request_attrs: t.Optional[t.List[str]] = None,
+    request_hook: t.Optional[RequestHook] = None,
+    response_hook: t.Optional[ResponseHook] = None,
+    tracer_provider: t.Optional[trace.TracerProvider] = None,
+):
+    if excluded_urls is None:
+        excluded_urls = _CONFIG_EXCLUDED_URLS
+    else:
+        excluded_urls = ExcludeList(excluded_urls)
+    traced_request_attrs = traced_request_attrs or _CONFIG_TRACED_REQUEST_ATTRS
+
+    _handle_request_wrapped = RequestHandler._handle_request
+
+    @functools.wraps(_handle_request_wrapped)
+    async def _handle_request_wrapper(
+        self, request: BaseRequest, start_time: float,
+    ):
+        if _is_suppressed(excluded_urls, request):
+            return await _handle_request_wrapped(self, request, start_time)
+
+        token = context.attach(propagate.extract(request.headers))
+
+        tracer = trace.get_tracer(__name__, __version__, tracer_provider)
```

Comment:
Creating a new tracer for every request is undesirable. Please create tracer once and use it.

Reply:
Good catch.